src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -361,18 +399,25 @@ private AIState clearStep()
     {
         WorkOrderBuild wo = job.getWorkOrder();
 
+        if(job.getStructure() == null)
+        {
+            //fix for bad structures
+            job.complete();
+        }
+
         if (wo.isCleared())
         {
             return AIState.BUILDER_STRUCTURE_STEP;
         }
 
-        BlockPos coordinates = job.getSchematic().getBlockPosition();
-        Block worldBlock = world.getBlockState(coordinates).getBlock();
+        BlockPos coordinates = job.getStructure().getBlockPosition();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'coordinates' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)